asdf/tags/core/ndarray.py

```diff
@@ -344,6 +346,16 @@ def __getattr__(self, attr):
             raise AttributeError()
         return getattr(self._make_array(), attr)
 
+    def __setitem__(self, *args):
+        # This workaround appears to be necessary in order to avoid a segfault
+        # in the case that array assignment causes an exception. The segfault
+        # originates from the call to __repr__ inside the traceback report.
+        try:
+            self._make_array().__setitem__(*args)
+        except Exception:
+            self._array = None
+            raise
```

Comment:
Since we've cut with 2.7:

```python
except Exception as e:
    self._array = None
    raise e from None
```

seems better - it does not pollute the error trace.